Big query connector (#48)
pachyderm/examples: bigquery/README.md

1. Go to the BigQuery web UI and create a new dataset.
2. In the GCP Console, go to the IAM & admin section and create a new [service account](https://console.cloud.google.com/iam-admin/serviceaccounts/).
3. Grant the service account the BigQuery Data Viewer and BigQuery Data Editor roles for the dataset you created in step 1.
4. Download the private key file for the service account and save it to a secure location.

### Usage
The underlying code is a Python script that ingests data from a BigQuery query and saves the results as a parquet file.

```
Create a parquet file from BigQuery using pandas-gbq

optional arguments:
  -h, --help            show this help message and exit
  -i INPUT_QUERY, --input_query INPUT_QUERY
                        Input query to run on BigQuery
  -o OUTPUT_FILE, --output_file OUTPUT_FILE
                        Output file path for the parquet file
  -p PROJECT_ID, --project_id PROJECT_ID
                        Google Cloud Project ID
  -c CREDENTIALS_FILE, --credentials_file CREDENTIALS_FILE
                        Google Cloud Service account file
```

> 1. Once you have a service account key, save it as `mycreds.json`.

Reviewer: ```suggestion
1. Once you have a service account key, save it with a descriptive name. In this example, we will use `gbq-pachyderm-creds.json`
```

2. Create the secret (be sure to add the namespace if your cluster is deployed in one).

```bash

> kubectl create secret generic gbqsecret --from-file=mycreds.json

Reviewer: ```suggestion
kubectl create secret generic gbqsecret --from-file=gbq-pachyderm-creds.json
```

```
With a namespace
```bash

> kubectl create secret generic gbqsecret --from-file=mycreds.json -n mynamespace

Reviewer: ```suggestion
kubectl create secret generic gbqsecret --from-file=gbq-pachyderm-creds.json -n mynamespace
```

```

3. Run the pipeline template with jsonnet.  

```bash
$ pachctl update pipeline --jsonnet gbq_ingest.jsonnet \
--arg inputQuery="SELECT country_name, alpha_2_code FROM bigquery-public-data.utility_us.country_code_iso WHERE alpha_2_code LIKE 'A%'" \
--arg outFile="gbq_output.parquet" \
--arg project="<project_name>" \

> --arg cronSpec="@every 30s"

Reviewer: ```suggestion
--arg cronSpec="* * 31 2 *" \
--arg credsFile="gbq-pachyderm-creds.json"
```

Can we use `"* * 31 2 *"` so that when a user copies and pastes this, it doesn't run? Then, the user can change the cronSpec to what they need or use `pachctl run cron gbq_ingest` to manually trigger the pipeline.

Author: Wouldn't we want it to run in the example? This is a small enough query that it should be okay, but we can definitely change it. Maybe we can add details in the readme or links to the docs to show other ways of running it?

```

## Configuring your own pipeline spec
You can configure your own pipeline spec with the secret by using these parameters in the pipeline spec. 

```json
"secrets": [ {
    "name": "gbqsecret",
    "mount_path": "/kubesecret/"}]
```
and
```json
    "env": {

>         "GOOGLE_APPLICATION_CREDENTIALS": "/kubesecret/mycreds.json"

Reviewer: ```suggestion
        "GOOGLE_APPLICATION_CREDENTIALS": "/kubesecret/gbq-pachyderm-creds.json"
```
